### PR TITLE
realtime 

### PR DESCRIPTION
**Pull request by econaxis**
integrates with realtime branch of trips-viewer

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            //     "provider": "gtfs-realtime",
+            //     "gtfs_realtime_url": "http://pt.mapstrat.com/current/gtfrealtime_VehiclePositions.bin"
+            // },
+            // {
+            //     "id": "barrie",
+            //     "provider": "gtfs-realtime",
+            //     "gtfs_realtime_url": "http://www.myridebarrie.ca/gtfs/GTFS_VehiclePositions.pb"
+            // }
+            // TODO: add GO transit
+        ]
```
Comment (econaxis) [inline src/index.ts:52]: Will add back once testing is done

File: kubernetes/orion-deployment.yaml [hunk 2]
```diff
@@ -36,36 +42,33 @@ spec:
...
-                    "provider": "gtfs-realtime",
-                    "gtfs_realtime_url": "http://www.myridebarrie.ca/gtfs/GTFS_VehiclePositions.pb"
-                  }
-                ]
-              }
+          - name: IS_SERVER
+            value: "true"
+        volumeMounts:
+          - name: my-data
+            mountPath: /data
```
Comment (EddyIonescu) [inline kubernetes/orion-deployment.yaml:49]: To clarify is this where the sqllite db is stored as to persist it?

File: migrations/001-vehicle-positions.sql [hunk 3]
```diff
@@ -0,0 +1,47 @@
+--------------------------------------------------------------------------------
+-- Up
+--------------------------------------------------------------------------------
+
+CREATE TABLE vehicle_position (
+      rid TEXT NOT NULL ,
+      vid TEXT NOT NULL,
+      lat TEXT NOT NULL,
+      lon TEXT NOT NULL,
```
Comment (EddyIonescu) [inline migrations/001-vehicle-positions.sql:9]: Wondering if it'd make sense to store as a float in sql?

File: migrations/001-vehicle-positions.sql [hunk 4]
```diff
@@ -0,0 +1,47 @@
+--------------------------------------------------------------------------------
+-- Up
+--------------------------------------------------------------------------------
+
+CREATE TABLE vehicle_position (
+      rid TEXT NOT NULL ,
+      vid TEXT NOT NULL,
```
Comment (EddyIonescu) [inline migrations/001-vehicle-positions.sql:7]: Wondering if any other types that use less space (like a varchar(120) or equivalent) would make sense here?

File: src/config.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        },
+        {
+            "id": "peterborough",
+            "provider": "gtfs-realtime",
+            "gtfs_realtime_url": "http://pt.mapstrat.com/current/gtfrealtime_VehiclePositions.bin"
+        },
+        {
+            "id": "barrie",
+            "provider": "gtfs-realtime",
+            "gtfs_realtime_url": "http://www.myridebarrie.ca/gtfs/GTFS_VehiclePositions.pb"
```
Comment (EddyIonescu) [inline src/config.ts:31]: Just to clarify any reason we don't need trip-updates from other agencies or is all the info we need from them in the vehicles one + GTFS?

**Review by EddyIonescu — COMMENTED** [review]
Just left some early minor comments on this. Will continue looking in more detail on the weekend/Monday, but don't expect any major comments.

File: migrations/001-vehicle-positions.sql (hunk 3 above)
Comment (econaxis) [reply migrations/001-vehicle-positions.sql:9]: I initially went with floats but I don't trust unique indexes with floats, so we just store everything as strings so unique comparisons can work.

File: migrations/001-vehicle-positions.sql (hunk 4 above)
Comment (econaxis) [reply migrations/001-vehicle-positions.sql:7]: Hmm I searched it up and according to https://stackoverflow.com/a/17798404 it's all the same on the backend.

File: src/config.ts (hunk 5 above)
Comment (econaxis) [reply src/config.ts:31]: Yep, as the realtime map is only for brampton transit right? We can add trip updates for other agencies, but right now I just didn't add it.

File: src/gtfs-parser.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+}
+function unixTimestampToSecondsOfDay(unixTimestamp, timezone) {
+    // timezone is from IANA timezone database, like "America/Toronto"
+    const torontoTime = moment.unix(unixTimestamp).tz(timezone);
+    return torontoTime.diff(torontoTime.clone().startOf('day'), 'seconds')
+}
+
+export function getClosestScheduledStopTime(delays: Record<string, number>, tripId: string, timestamp: number) {
+
+    const stop_times = getStoptimes({trip_id: tripId})
```
Comment (EddyIonescu) [inline src/gtfs-parser.ts:29]: Could you add a comment saying what this does - from what I see it iterates through the delay for each stop and return the first scheduled time at a stop at or past the current time plus the delay from the TripUpdates feed.

**Review by EddyIonescu — APPROVED** [review]
Looks good! Just left some minor comments, then feel free to merge.

**Pull request merged**